readthedocs/projects/views/public.py

```diff
@@ -208,14 +210,18 @@ def project_download_media(request, project_slug, type_, version_slug):
     )
     privacy_level = getattr(settings, 'DEFAULT_PRIVACY_LEVEL', 'public')
     if privacy_level == 'public' or settings.DEBUG:
-        path = os.path.join(
+        storage_path = version.project.get_storage_path(type_=type_, version_slug=version_slug)
+        if storage.exists(storage_path):
+            return HttpResponseRedirect(storage.url(storage_path))
+
+        media_path = os.path.join(
```

Comment:
Believe this logic is actually what the DefaultFileStorage will do, so I don't think this will ever get hit.

Reply:
Good point. 👍